### DISCUSSION TITLE
Some improvement ideas: options, dowload instructions and additional scripts

### DISCUSSION THREAD

**Discussion by h00g** [007revad/Synology_HDD_db#211] [Ideas]
These ideas stem from a first impression after getting directed to try this script when asking for advice in a Synology discussion.

Sometimes such first impressions can seem confused or negative, sometimes a developer with full insight may have the immediate thought "... but that's obvious" or it can lead to a positive change improving on minor things that will clarify for other newcomers. 

### options
Not [all options](https://github.com/007revad/Synology_HDD_db/blob/main/README.md#options) are explained making it hard to know when to use them, when to avoid setting the flag and what benefit/utility or drawback/risk it will lead to.
E.g.

- What are the implications of setting --m2 or not and in which cases would that be a benefit or drawback?
- The same for the flag --wdda.

If there already exists a more thorough description then adding a link to that could be a solution.

### download instructions
The dowload instruction [text](https://github.com/007revad/Synology_HDD_db/blob/main/README.md#download-the-script) and [image](https://github.com/007revad/Synology_HDD_db/blob/main/images/how_to_download.png) could, with likely low effort, be updated to bring clarity.

1. The top of the [image](https://github.com/007revad/Synology_HDD_db/blob/main/images/how_to_download.png) show the names [syno_hdd_db.sh](https://github.com/007revad/Synology_HDD_db/blob/main/syno_hdd_db.sh) and [README.md](https://github.com/007revad/Synology_HDD_db/blob/main/README.md) but only the [README.md](https://github.com/007revad/Synology_HDD_db/blob/main/README.md) page show the release label taking you to [https://github.com/007revad/Synology_HDD_db/releases](https://github.com/007revad/Synology_HDD_db/releases)
2. Adding a link to [https://github.com/007revad/Synology_HDD_db/releases](https://github.com/007revad/Synology_HDD_db/releases) from the [dowload instruction](https://github.com/007revad/Synology_HDD_db/blob/main/README.md#download-the-script) could be a quick fix.
3. Is dowloading the whole repository needed? Maybe the dowload instruction [text](https://github.com/007revad/Synology_HDD_db/blob/main/README.md#download-the-script) could easily be augmented with e.g. "For a minimal download you'll just need to download [https://github.com/007revad/Synology_HDD_db/blob/main/syno_hdd_db.sh](https://github.com/007revad/Synology_HDD_db/blob/main/syno_hdd_db.sh) and [syno_hdd_vendor_ids.txt](https://github.com/007revad/Synology_HDD_db/blob/main/syno_hdd_vendor_ids.txt) and place them in the same folder on your NAS"

### additional scripts
This repository also refers to [my other scripts](https://github.com/007revad/Synology_HDD_db/blob/main/my-other-scripts.md).

Could the info be extended to more clearly map the needs to which (one or several) of the scripts that would do what?

In my case I have a DS423+ with DSM 7.2 supporting using M.2 drives as volumes but I have two Gigabyte M.2 drives that are not in Synology's compatibility list.
Would I need to use [Synology_HDD_db](https://github.com/007revad/Synology_HDD_db) and/or [Synology_enable_M2_volume](https://github.com/007revad/Synology_enable_M2_volume) and/or [Synology_M2_volume](https://github.com/007revad/Synology_M2_volume)?

**Comment by 007revad** [comment]
@h00g 

Thanks for the feedback Lars. 

With a DS423+ using DSM 7.2 you only need the Synology HDD db script.

### options 
1. I've added information about wdda in bullet point 11 in the [What the script does](https://github.com/007revad/Synology_HDD_db#what-the-script-does) section and in the [Options when running the script](https://github.com/007revad/Synology_HDD_db#options-when-running-the-script-) section.
2. I've added information about the -f or --force option in the **Notes** in the [Options when running the script](https://github.com/007revad/Synology_HDD_db#options-when-running-the-script-) section.
3. I've removed all reference to the -m or --m2 option. It was added in the first version of the  script in case someone had manually created an NVMe volume via SSH and was concerned that the script may cause their NVMe volume to become inaccessible. I don't believe anyone uses the -m or --m2 option but I've left it in the script just in case.

### download instructions
I've updated the [Download the script](https://github.com/007revad/Synology_HDD_db#download-the-script) section.

### additional scripts
The **Synology HDD db** versus **Synology enable M2 volume** versus **Synology M2 volume** has changed a lot over the last year.

- Any 2020 or newer Synology model using DSM 7.2 or later only needs Synology HDD db.
- Anyone using Synology enable M2 volume or Synology M2 volume also needs Synology HDD db.
- Synology enable M2 volume is in my opinion no longer needed because Synology HDD db achieves the same result (but in a different way).
- Synology M2 volume is only needed for older Synology models or older DSM versions.

The **Synology enable M2 volume** page includes a list and a table showing the [differences between Synology enable M2 volume and Synology M2 volume](https://github.com/007revad/Synology_enable_M2_volume#how-is-this-different-to-the-synology_m2_volume-script)
